### ISSUE TITLE
WebSocket manager

### ISSUE THREAD

**Issue by HRDepartment** [po-devs/webclient#65]
Basically an internal "library" used to manage the websocket. As more and more network commands are added, management will become a pain (especially at the point where you want to optimize the client as much as possible).

A global object with some high level functions and an enum containing the commands, more or less.

If no one else is interested in doing this I might put some time into it.

**Comment by HRDepartment** [comment]
Is this a priority at all at the moment? The earlier it's done the less code has to be rewritten, but it may not necessary just yet.

**Comment by coyotte508** [comment]
I don't understand what you want to do.

What's wrong with the way it's done now? Also most commands (75%) are already done.

**Comment by HRDepartment** [comment]
A class that manages the websocket (.send, state, etc), including commands sent to/from the webclient.

Currently, everything is handled in one function, and other random parts here and there.

_Eventually_, you'll want to optimize the protocol, and this makes it fairly easy to do so.

tl;dr _something_ (no radical changes, everything should remain plain-text) like the Network and Analyzer classes.

**Comment by coyotte508** [comment]
The network class is obsolete since its job is already done with the websocket.

Commands could be directly dealt with with dealWithXxxx (like how battle commands are handled), but it's really not a priority.

**Comment by HRDepartment** [comment]
No, I meant something like this

```
Socket.sendCommand(Socket.commands.Login /* or 'Login', etc */, {} /* basically some kind of payload, clearly objects will be string-ified*/); 
```

instead of

```
websocket.send("login|" + JSON.stringify(payload));
```

Later on, when the network should be optimized, `Socket.commands.Login` could be an integer (or just something short like `l`) instead of a string, as well as some minor opt-in optimizations.

**Comment by coyotte508** [comment]
No it won't be. Things are jsoned, it's for clarity of the protocol, it'll stay text, not integers.

Enums in javascript don't really work, strings are commonly used instead.

Anyway, that's really not a major concern atm.
